src/BigT/bigt.java

```diff
@@ -0,0 +1,509 @@
+            DeleteRecException, IndexSearchException, IteratorException, LeafDeleteException, InsertException,
+            IOException, DeleteFashionException, LeafRedistributeException, RedistributeException, InsertRecException,
+            FreePageException, RecordNotFoundException, IndexFullDeleteException {
+        boolean status;
+        if(type == 1){
+            status = true;
+        }else if(type == 2){
+            status = indexFiles.get(2).Delete(new StringKey(map.getRowLabel()), mid);
+        }else if(type == 3){
+            status = indexFiles.get(3).Delete(new StringKey(map.getColumnLabel()), mid);
+        }else if (type == 4){
+            status = indexFiles.get(4).Delete(new StringKey(map.getColumnLabel() + "%" + map.getRowLabel()), mid);
+        }else{
+            status = indexFiles.get(5).Delete(new StringKey(map.getRowLabel() + "%" + map.getValue()), mid);
+        }
+        return status;
+    }
+
+    public void insertIndexUtil(MID mid, Map map, int heapFileIndex)
+            throws KeyTooLongException, KeyNotMatchException, LeafInsertRecException, IndexInsertRecException,
+            ConstructPageException, UnpinPageException, PinPageException, NodeNotMatchException, ConvertException,
+            DeleteRecException, IndexSearchException, IteratorException, LeafDeleteException, InsertException,
+            IOException{
+        utilityIndex.insert(new StringKey(map.getRowLabel() + map.getColumnLabel() + "%" + map.getTimeStamp() + "%" + heapFileIndex), mid);
+    }
+
+    public int getMapCnt() throws InvalidSlotNumberException, InvalidTupleSizeException, HFDiskMgrException,
+            HFBufMgrException, IOException {
+        int totalMapCount = 0;
+        for(int i = 1; i <= 5; i++){
+            totalMapCount += heapFiles.get(i).getRecCntMap();
+        }
+        return totalMapCount;
+    }
+
+    public int getRowCnt()  throws Exception{
+        return getCount(3);
+    }
+
+    public int getColumnCnt()  throws Exception{
+        return getCount(4);
+    }
+
+    public int getCount(int orderType) throws Exception{
+        int numBuf = (int)((SystemDefs.JavabaseBM.getNumBuffers()*3)/4);
+        Stream stream = new Stream(this.name, 1,  "*", "*", "*", numBuf);
+        Map t = stream.getNext();
+        int count = 0;
+        String temp = "\0";
+        while(t != null) {
+            t.setFldOffset(t.getMapByteArray());
+            if(orderType==3){
+                if(!t.getRowLabel().equals(temp)){
+                    temp = t.getRowLabel();
+                    count++;
+                }
+            }else{
+                if(!t.getColumnLabel().equals(temp)){
+                    temp = t.getColumnLabel();
+                    count++;
+                }
+            }
+            t = stream.getNext();
+        }
+        stream.closestream();
+        return count;
+    }
+
+
+    public MID insertMap(Map map, int type) throws HFDiskMgrException,
```

Comment:
MID insertMap(byte[] mapPtr) Insert map into the big table, return its Mid. The insertMap() method ensures that there are at most three maps with the same row and column la- bels, but different timestamps, in the bigtable. When a fourth is inserted, the one with the oldest label is dropped from the big table.